aptos-move/framework/src/natives/code.rs

```diff
@@ -203,6 +205,21 @@ pub struct NativeCodeContext {
     requested_module_bundle: Option<PublishRequest>,
 }
 
+impl VersionControlledNativeExtension for NativeCodeContext {
+    fn undo(&mut self) {
+        // No-op: nothing to undo.
+    }
+
+    fn save(&mut self) {
+        // No-op: nothing to save.
+    }
+
+    fn update(&mut self, _txn_hash: &[u8; 32], _script_hash: &[u8], _session_counter: u8) {
+        self.enabled = true;
```

Comment:
Note: we probably want to disable module publish for AA: currently one can run `code::publish_txn` in prologue via AA, but AptosVM will ignore these.

Reply:
Is this related to the above TODO? 
How would we know it's the prologue - with session counter?
Maybe it's worth disabling so we can make sure it passes everything when testing the new sessions.

But actually why does enabled matter at extract_publish_request, but not already at native_request_publish? Wouldn't it make sense to not even set requested_module_bundle when not enabled? (or even return some invariant error)

Reply:
`enabled` matter for `extract_publish_request` because we have `init_module` and we want to make sure `init_module` does not publish code (it cannot, but users can run this partially)